### ISSUE TITLE
python segmentation.py  demo fails

### ISSUE THREAD

**Issue by rodrigob** [meng-tang/KernelCut#4]
Just trying the example command line
`$ python segmentation.py -i 124084.jpg -b 10 20 300 300 -k 100`

Got the following error

```
image directory is  some_path/KernelCut/PyKernelCut
image name is  124084.jpg
hardconstraintsflag False
computing k nearest neighbors...
3.623304 seconds process time
Traceback (most recent call last):
  File "segmentation.py", line 67, in <module>
    main()
  File "segmentation.py", line 60, in main
    segmentation = KernelCut(image, opt)
  File "some_path/KernelCut/PyKernelCut/kernelcut.py", line 124, in KernelCut
    affinity_matrix = ConstructKNNGraph(image, opt.KNN_K, opt.xyscale)
  File "some_path/KernelCut/PyKernelCut/kernelcut.py", line 100, in ConstructKNNGraph
    affinity_matrix = (affinity_matrix + affinity_matrix.transpose(True)) /2
  File "some_other_path/virtualenvs/pykernelcut/local/lib/python2.7/site-packages/scipy/sparse/csr.py", line 133, in transpose
    raise ValueError(("Sparse matrices do not support "
ValueError: Sparse matrices do not support an 'axes' parameter because swapping dimensions is the only logical permutation.
```

This is probably related to different scipy versions.
I have in my virtualenv (from a fresh install from a few minutes ago)

```
$ pip list
appdirs (1.4.0)
cycler (0.10.0)
dask (0.13.0)
decorator (4.0.11)
functools32 (3.2.3.post2)
matplotlib (2.0.0)
networkx (1.11)
numpy (1.12.0)
olefile (0.44)
packaging (16.8)
Pillow (4.0.0)
pip (9.0.1)
PyMaxflow (1.2.4)
pyparsing (2.1.10)
python-dateutil (2.6.0)
pytz (2016.10)
scikit-image (0.12.3)
scikit-learn (0.18.1)
scipy (0.18.1)
setuptools (34.1.1)
six (1.10.0)
subprocess32 (3.2.7)
toolz (0.8.2)
wheel (0.29.0)
```

if a specific library version is needed, that can be specified in the requirements.txt file (see pull request #3 ).

**Comment by rodrigob** [comment]
Fix suggested in #5; with it the demonstrations run fine.

**Comment by meng-tang** [comment]
affinity_matrix.transpose(True) to affinity_matrix.transpose(copy=True) should fix
